### ISSUE TITLE
Preview of the Repository

### ISSUE THREAD

**Issue by mintbomb27** [IEEE-VIT/templa-rs#5]
The preview of the repository can be shown while the user browses through the available templates on another block aside the search results.

The available files and folders in the repository shown as a tree would suffice.

You can either scrape the repository or recursively clone the repositories. Any better method is welcome!

**Comment by kugiyasan** [comment]
Can I take this issue?

I played around a bit and I'll try to scrape the github page, since it seems like it takes roughly the same time as a shallow clone.
```sh
$ time git clone --quiet https://github.com/h5bp/html5-boilerplate
real	0m7.366s
user	0m0.917s
sys	0m0.448s
$ time curl -sL https://github.com/h5bp/html5-boilerplate -o a.html
real	0m0.698s
user	0m0.029s
sys	0m0.017s
$ time git clone --quiet --depth 1 https://github.com/h5bp/html5-boilerplate
real	0m0.562s
user	0m0.030s
sys	0m0.020s
```

**Comment by mintbomb27** [comment]
Closed by #42
